extensions/vscode/webviews/homeView/src/components/views/Credentials.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <TreeSection title="Credentials" :actions="sectionActions">
+    <WelcomeView v-if="home.sortedCredentials.length === 0">
+      <p>No credentials have been added yet.</p>
+    </WelcomeView>
+    <TreeItem
+      v-else
+      v-for="credential in home.sortedCredentials"
+      :title="credential.name"
+      :description="credential.url"
+      :codicon="
+        credential.guid === CredentialGUIs.EnvironmentGUID
+          ? 'codicon-bracket'
+          : 'codicon-key'
+      "
+      align-icon-with-twisty
+      :data-vscode-context="
+        credential.guid === CredentialGUIs.EnvironmentGUID
+          ? undefined
+          : `{&quot;webviewSection&quot;: &quot;credentials-tree-item&quot;, &quot;credentialGUID&quot;: &quot;${credential.guid}&quot;, &quot;credentialName&quot;: &quot;${credential.name}&quot;, &quot;preventDefaultContextMenuItems&quot;: true}`
+      "
```

Comment:
Might you consider moving this into a variable but build it up and then encode it, so it becomes easier to understand what it is comprised of?

Reply:
I tried that a few times in the previous usage of `data-vscode-context` and never got it right. I'll time box it a bit here and try to get it working since this is _very_ difficult to read.

Reply:
Was able to get this working in https://github.com/posit-dev/publisher/pull/2151/commits/465033582463bfb246e39b17d5f3f2418a7dca5c

I'll create a follow-up PR to do the same in our other usage of `data-vscode-context`